src/main/java/org/myrobotlab/service/Pcf8574.java

```diff
@@ -499,23 +492,17 @@ public void onRegistered(Registration s) {
 
   public void pinMode(int address, int mode) {
     PinDefinition pinDef = getPin(address);
-    if (mode == INPUT) {
-      pinDef.setMode("INPUT");
-      directionRegister &= ~(1 << address);
-    } else {
-      pinDef.setMode("OUTPUT");
-      directionRegister |= (1 << address);
-    }
+    // There is no direction register in the PCF8574 it is always BIDRECTIONAL.

```

Comment:
might be worth logging a warning or info - saying if its != "BIDIRECTIONAL"
Pcf8574 only does BIDIRECTIONAL